### PR TITLE
Changed RecipeYamlParser to the Resource type.

### PR DESCRIPTION
**Pull request by kitskub**
This is at the request of @royawesome.

Also fixed spacing in recipes.yml

Signed-off-by: Jack Huey kitskub@gmail.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/VanillaPlugin.java [hunk 1]
```diff
@@ -1,249 +1,253 @@
-/*
```
Comment (ZNickq) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java:1]: What happened here? o.O

**Comment by kitskub** [comment]
I'm sorry, but that whole won't file reset at all. I have all my settings right, but git is not doing what I want.

**Comment by lukespragg** [comment]
It's actually what it's supposed to be now. The line endings are supposed to be LF on the repo, but somehow this file was CRLF on the repo here at GitHub. So it's all good and shouldn't happen again. :)

Did you make any actual changes to VanillaPlugin.java?

File: src/main/java/org/spout/vanilla/VanillaPlugin.java (hunk 1 above)
Comment (lukespragg) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java:1]: See comment on pull request. ;)

**Comment by kitskub** [comment]
Yes, I added two lines. I'll mark them.

File: src/main/java/org/spout/vanilla/VanillaPlugin.java [hunk 2]
```diff
@@ -1,249 +1,253 @@
...
+	}
+
+	@Override
+	public void onLoad() {
+		instance = this;
+		engine = getEngine();
+		config = new VanillaConfiguration(getDataFolder());
+		Protocol.registerProtocol("VanillaProtocol", new VanillaProtocol());
+		Spout.getFilesystem().registerLoader("mappalette", new MapPaletteLoader());
+		Spout.getFilesystem().registerLoader("recipe", new RecipeLoader());
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java:136]: This one.

File: src/main/java/org/spout/vanilla/VanillaPlugin.java [hunk 3]
```diff
@@ -1,249 +1,253 @@
...
+			}
+
+			((Server) engine).bind(new InetSocketAddress(split[0], port), new VanillaBootstrapProtocol());
+		} else if (engine.getPlatform() == Platform.CLIENT) {
+			//TODO if (engine instanceof Client) do stuff? | No, check engine.getPlatform()
+		}
+
+		VanillaMaterials.initialize();
+		MapPalette.DEFAULT = (MapPalette) Spout.getFilesystem().getResource("mappalette://Vanilla/resources/map/mapColorPalette.dat");
+		RecipeYaml.DEFAULT = (RecipeYaml) Spout.getFilesystem().getResource("recipe://Vanilla/resources/recipes.yml");
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/VanillaPlugin.java:156]: And this one.

**Pull request merged**